charts/vault-csi-provider/templates/daemonset.yaml

```
@@ -30,3 +30,5 @@ spec:
         - name: providervol
           hostPath:
             path: {{ .Values.csiProviderPath }}
+      nodeSelector:
+        beta.kubernetes.io/os: linux
```

Is Karbon still using this with a beta prefix? `beta.kubernetes.io/os` is deprecated according to the [docs](https://kubernetes.io/docs/reference/labels-annotations-taints/) and `kubernetes.io/os` should be used instead.
```suggestion
        kubernetes.io/os: linux
```

---

It's the same on the upstream secrets-store-csi-driver-provider-vault: https://github.com/hashicorp/secrets-store-csi-driver-provider-vault/blob/master/deployment/provider-vault-installer.yaml#L72

---

`beta.` was deprecated in 1.14 and the csi-provider needs at least 1.15 - I think we can safely use `kubernetes.io/os`

---

In any case (independant of if there is a beta beta prefix or not), this nodeSelector should be confgured via values and not be a hardcoded magic string in the Daemonset.

---

Hardcoding a `nodeSelector` is acceptable when we know that the daemonset can't be executed on Windows nodes, but we should probably provide an option to extend the `nodeSelector` or configure our default in `values.yaml`.

---

what has this to do  with Karbon?

---

It's also just taken from upstream here.

---

https://github.com/hashicorp/secrets-store-csi-driver-provider-vault/blob/master/deployment/provider-vault-installer.yaml